lighthouse-core/audits/first-meaningful-paint.js

```diff
@@ -120,27 +120,14 @@ class FirstMeaningfulPaint extends Audit {
   }
 
   /**
-   * @param {!Array<!Object>} traceData
+   * @param {!Object} tabTrace
    */
-  static collectEvents(traceData) {
-    // Parse the trace for our key events and sort them by timestamp.
-    const events = traceData.filter(e => {
-      return e.cat.includes('blink.user_timing') || e.name === 'TracingStartedInPage';
-    }).sort((event0, event1) => event0.ts - event1.ts);
-
-    // The first TracingStartedInPage in the trace is definitely our renderer thread of interest
-    // Beware: the tracingStartedInPage event can appear slightly after a navigationStart
-    const startedInPageEvt = events.find(e => e.name === 'TracingStartedInPage');
-    // Filter to just events matching the frame ID for sanity
-    const frameEvents = events.filter(e => e.args.frame === startedInPageEvt.args.data.page);
-
-    // Find our first FCP
-    const firstFCP = frameEvents.find(e => e.name === 'firstContentfulPaint');
-    // Our navStart will be the latest one before fCP.
-    const navigationStart = frameEvents.filter(e =>
-        e.name === 'navigationStart' && e.ts < firstFCP.ts).pop();
+  static collectEvents(tabTrace) {
+    const navigationStart = tabTrace.navigationStartEvt;
+    const firstFCP = tabTrace.firstContentfulPaintEvt;
+
     // fMP will follow at/after the FCP, though we allow some timestamp tolerance
-    const firstMeaningfulPaint = frameEvents.find(e =>
+    const firstMeaningfulPaint = tabTrace.traceEvents.find(e =>
```

Comment:
FCP is super reliable in a trace.  I use FCP to select the best navStart and so far that is the most robust method.  (if this werent the case, i wouldnt have grabbed FCP there)

FMP is definitely not (and im thinking we should soon write our own FMP selection based on the candidates, as the real one's network idle detection is taking too long. anyway..)

however technically i could do this FMP selection in the computed artifact just so that TTI audit doesn't need to depend on both of the taboftrace artifact and fmp audit at the same time.  that's fine with me.